server/app/util/filter.py

```diff
@@ -35,36 +49,47 @@ def parse_filter(filter, schema):
     The results is a matrix with the cells the pass the filter and at this point all the genes
     :param filter: flask's request.args
     :param schema: dictionary schema
+    :raises QueryStringError
     :return:
     """
     query = {}
+
     for key in filter:
-        value = filter.getlist(key)
-        if key not in schema:
-            raise QueryStringError("Error: key {} not in metadata schema".format(key))
-        query[key] = {
-            "variable_type": schema[key]["variabletype"],
-            "value_type": schema[key]["type"]
-        }
-        if query[key]["variable_type"] == "categorical":
-            query[key]["query"] = [_convert_variable(query[key]["value_type"], v) for v in value]
-        elif query[key]["variable_type"] == "continuous":
-            value = value[0]
+        axis, annotation = key.split(":", 1)
+        if axis not in [ax for ax in Axis]:
+            raise QueryStringError(f"Error: bad axis in query string {axis}")
+        if axis not in query:
```

Comment:
consider [collections.defaultdict](https://docs.python.org/3/library/collections.html#collections.defaultdict) (i would not necessarily adopt it, but you can look at it and see if you think it's right for you)

Reply:
Added that. Good catch, I had meant to use default dict and forgot.